lib/R/Mbio/alpha_div.R

```diff
@@ -0,0 +1,65 @@
+#### Docs TODO
+#### Compute alpha diversity
+alphaDiv <- function(otu, method = c('shannon','simpson','evenness'), verbose = c(TRUE, FALSE)) {
+
+    # Initialize and check inputs
+    method <- plot.data::matchArg(method)
+    verbose <- plot.data::matchArg(verbose)
+
+    computeMessage <- ''
+    plot.data::logWithTime(paste("Received OTU table with", NROW(otu), "samples and", (NCOL(otu)-1), "taxa."), verbose)
+    
+
+    # Compute alpha diversity
+    if (identical(method, 'shannon') | identical(method, 'simpson')){
+      alphaDivDT <- try(vegan::diversity(otu[, -c('SampleID')], method))
+    } else if (identical(method, 'evenness')) {
+      alphaDivDT <- try(vegan::diversity(otu[, -c('SampleID')], 'shannon') / log(vegan::specnumber(otu)))
+    }
+
+    if (is.error(alphaDivDT)) {
+      computeMessage <- paste('Error: alpha diversity', method, 'failed')
+      # Also handle dt, results? Or just stop?
+    } else {
+      computeMessage <- paste('Computed', method, 'alpha diversity.')
+    }
+
+    # Collect results
+    dt <- data.table('SampleID'= otu[['SampleID']],
+                      'alphaDiv' = alphaDivDT)
+    data.table::setnames(dt, 'alphaDiv', method)
+
+    results <- list(
+      'dt' = dt,
```

Comment:
you might consider trying to standardize the format of this results list. maybe a 'data', 'stats', and 'metadata' entry. and those can each be a data.table or list themselves as necessary. then we could consider having a single write* helper that knows to look for each of those entries in the results list. (this is approaching oo territory, but we can punt on that for now)